### DISCUSSION TITLE
lpcxpresso55s16 zephyr undefined reference to `__aeabi_d2iz'

### DISCUSSION THREAD

**Discussion by Aravinthkalai** [zephyrproject-rtos/zephyr#58295] [Q&A] [answered]
Hello Everyone !

I'm using lpcxpresso55s16 Eval Board with zephyr stack, I just created a thread which calling one function , In that function I'm performing some floating point calculation based on the result I'm incrementing the static variable when I try to compile it's compiling but in linker stage it's throwing error like undefined reference to `__aeabi_d2iz', I'm struggling to fix this issue , if you have any suggestion or solution please reply to this query.

Details :

board : lpcxpresso55s16

stack : zephyr 3.3.99
//File Name : abc.c
#include <kernel.h>
#include <def.h>
static struct k_thread handler_thread_data;

K_THREAD_STACK_DEFINE(handler_thread_stack,0X400);

void create_thread()
{

task_tid = k_thread_create(&handler_thread_data,handler_thread_stack,K_THREAD_STACK_SIZEOF(handler_thread_stack),handler_thread,NULL,NULL,NULL,2,0,K_FOREVER);
	if(!task_tid)
	{
		return task_init_failed;
	}
	k_thread_name_set(task_tid,"data_task");
	initialized = true;
}

void handler_thread(void *arg1,void *arg2, void *arg3)
{

	for(;;)
	{

	    calc();
            k_msleep(50);
    }
}

//Filename : def.c
#include <def.h>	
static double c=8.725;
static uint16_t inc=0;
void calc()
{
    
    double a;
    long b;
    a=10.5;
    b= (long)(a * c);  // compiler pointing this line and says undefined reference to `__aeabi_d2iz'

    if (b > 40) 
    {
          inc++;
    }	
    if(inc>1000)
    {
        inc=0;
    } 
}
``

**Comment by DerekSnell** [comment]
Sorry @Aravinthkalai , I don't see the issue.

I modified the [hello_world sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/hello_world/src/main.c) with the attached [main.c](https://github.com/zephyrproject-rtos/zephyr/files/11575626/hello_world.zip) using your code snippets, and it builds without errors for me.  I am using the latest `main` branch (commit dd8a65c03ce055d74ca0a2e188fedec24ad81e07).    

I see that `__aeabi_d2iz` converts a double to an int.  If this is missing in your build, it means the linker is not including it from the runtime library for some reason.  Are you setting `CONFIG_FPU`?  I have had issues in the past with missing floating point APIs, and it was related to the [Floating point ABI](https://github.com/zephyrproject-rtos/zephyr/blob/7457bcf0a2b3d3523b8b3ea971771337f3751f1f/arch/arm/core/aarch32/Kconfig#L260), which can be configured with Kconfigs.  Maybe that will help.

Let us know what you find

**Reply by Aravinthkalai** [reply, marked as answer]
@DerekSnell Thanks for your reply, actually this is not my actual code , I mocked my code and shared here , my issue is solved , I wrote the long calculation in a single line , now I split the long calculation in line by line and did proper type casting now it's works fine for me, once again thank you for your reply.
